models/src/main/scala/coop/rchain/models/rholang/sort/Sortable.scala

```diff
@@ -0,0 +1,25 @@
+package coop.rchain.models.rholang.sort
+import coop.rchain.models.Expr.ExprInstance
+import coop.rchain.models.Expr.ExprInstance.GBool
+import coop.rchain.models._
+
+trait Sortable[T] {
```

Comment:
I still don't think we need this type class. You can test that terms are sorted separately from `assertSpatialMatch`. 

```scala
def assertSorted(par: Par): Assertion = ParSortMatcher.sortMatch(par) == par

test("some test") {
assertSorted(target)
assertSorted(pattern)
assertSpatialMatch(target, pattern, expectedResult)
```

I don't like the fact that we are creating a type class only for the tests.

Reply:
Most of `target`-s and `pattern`s aren't Pars, so I'd have to pattern match over types not having a thing in common (on type level), and I'd have to put each one of those into a Par in a different way (a Send to sends field, an Expr to expressions field, etc...) and sort the Par, or just call the relevant `*SortMatcher.sortMatch` - which the typeclass does for me in a way better, cleaner and less error-prone way than both the other approches. 

Additionally, the `Sortable` typeclass captures and solidifies in types a very strong convention we have. I'd dare say we already have that typeclass in `dev`, just on convention level and can't benefit from it fully due to that. E.g.:
 - the compiler doesn't understand that convention and doesn't help
 - there's not a way to do sth for anything sortable, like I need to do above and wanted many times during debugging so far
 - newcomers have to discover the convention, while a typeclass would make things obvious
 - without that typeclass, it's easy to break the convention, because no types guard it

I'm sure replacing all the FooSortMatch.sortMatch with Sortable.sortMatch would make the code more readable, especially to those not knowing of the SortMatcher convention/typeclass yet (I've been that person a month ago). I plan to do that in a later PR.

Reply:
And since #1145 is ready to merge, you go to dev first and I'll update my PR :)